lib/csv.rb

```diff
@@ -1886,16 +1903,37 @@ def table(path, **options)
     private_constant :ON_WINDOWS
 
     private
-    def may_enable_bom_deletection_automatically(mode, options, file_opts)
-      # "bom|utf-8" may be buggy on Windows:
-      # https://bugs.ruby-lang.org/issues/20526
-      return if ON_WINDOWS
+    def may_enable_bom_detection_automatically(filename_or_io,
+                                               mode,
+                                               options,
+                                               file_opts)
+      if filename_or_io.is_a?(StringIO)
+        # Support to StringIO was dropped for Ruby 2.6 and earlier without BOM support:
+        # https://github.com/ruby/stringio/pull/47
+        return if RUBY_VERSION < "2.7"
+      else
+        # "bom|utf-8" may be buggy on Windows:
+        # https://bugs.ruby-lang.org/issues/20526
+        return if ON_WINDOWS
+      end
       return unless Encoding.default_external == Encoding::UTF_8
       return if options.key?(:encoding)
       return if options.key?(:external_encoding)
       return if mode.include?(":")
       file_opts[:encoding] = "bom|utf-8"
     end
+
+    if RUBY_VERSION < "2.7"
+      def init_stringio!(str, mode, opts)
+        opts.reject! { |k, _| k == :universal_newline || DEFAULT_OPTIONS.key?(k) }
```

Comment:
Could you use this style to align with the existing code?

```suggestion
        opts.delete_if {|k, _| k == :universal_newline or DEFAULT_OPTIONS.key?(k)}
```

Reply:
Thank you for the feedback. I've followed all your suggestions.